epoptes-client/lock-screen

```diff
@@ -23,78 +24,73 @@
-
-        imagepb = gtk.gdk.pixbuf_new_from_file_at_size('lock.svg', 250, 250)
-        image = gtk.Image()
-        image.set_from_pixbuf(imagepb)
-        vbox.pack_start(image, True)
-
-        # Italic fonts are trancated in the sides, so add a space
-        self.label = gtk.Label(" " + msg + " ")
-        self.label.modify_font(pango.FontDescription('FreeSans italic 18'))
-        self.fade(True)
-        vbox.pack_start(self.label, True)
+        frontview = Gtk.Window()
+        frontview.resize(swidth, sheight)
 
-        align = gtk.Alignment(0.5, 0.5, 0, 0)
-        align.add(vbox)
-        frontview.add(align)
+        box = Gtk.Box(
+            orientation=Gtk.Orientation.VERTICAL, spacing=smin/12, halign=Gtk.Align.CENTER, valign=Gtk.Align.CENTER)
+        image = Gtk.Image(pixbuf=GdkPixbuf.Pixbuf.new_from_file_at_size('lock.svg', smin/3, smin/3))
+        box.pack_start(image, False, False, 0)
+        self.label = Gtk.Label(label=msg, name="black")
+        box.pack_start(self.label, False, False, 0)
+        frontview.add(box)
 
         backlock.show_all()
         frontview.show_all()
 
         frontview.set_keep_above(True)
         frontview.fullscreen()
-        gdk.beep()
-        gdk.keyboard_grab(backlock.window, False, 0L)
-
-        # To automatically unlock the screen after 7 seconds, uncomment this:
-        #gobject.timeout_add(7000, self.unlock)
-
-
-    def fade(self, firsttime=False):
-        if firsttime:
-            self.fadesecs = 3.0
-            self.fadestart = time.time()
-            # For debugging:
-            #print self.fadestart, self.fadesecs
+        Gdk.beep()
+        Gdk.keyboard_grab(backlock.get_window(), False, 0)
 
-        percentage = (time.time() - self.fadestart) / self.fadesecs
-        i = min(round(255 * percentage), 255)
-        c = "#%02x%02x%02x" % (i, i, i)
-        # For debugging:
-        #print time.time(), '-', 100 * percentage, "completed, rendering color ", c
-        self.label.modify_fg(gtk.STATE_NORMAL, gdk.Color(c))
-        # Put a sleep() here if you want to test for slow thin clients:
-        #time.sleep(0.6)
-        if i < 255:
-            gobject.timeout_add(60, self.fade)
+        # Transitions need an event to start
+        GLib.timeout_add(100, self.do_transition)
 
-        return False
+        # While developing, run e.g. `./lock-screen "" 5` to only lock for 5 seconds
+        if unlock_secs is not None:
+            GLib.timeout_add(unlock_secs*1000, self.unlock)
 
+    def do_transition(self):
+        self.label.set_name("white")
 
     def unlock(self):
-        gdk.keyboard_ungrab(0L)
+        Gdk.keyboard_ungrab(0)
         exit()
 
 
-if len(sys.argv) <= 1:
-    lockscreen().lock()
-else:
-    lockscreen().lock(sys.argv[1])
-gtk.main()
+LockScreen().lock(
+    sys.argv[1] if len(sys.argv) > 1 and sys.argv[1] else "The screen is locked by a system administrator.",
+    int(sys.argv[2]) if len(sys.argv) > 2 else None
```

Comment:
I'm guilty of enjoying the one-line boolean operations in my code too, but they make the code less readable! I would suggest to try to avoid them.